packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -414,7 +421,19 @@ export class StripeHelper {
       return;
     }
 
-    if (customer.metadata.userid !== uid) {
+    // Since the uid is just metadata and it isn't required when creating a new
+    // customer _on Stripe dashboard_, we have an edge case where the customer
```

Comment:
Nice catch on this case! While our instructions indicate the new record should have the userid set in metadata, I'm sure someone will forget to add it at some point before going back and updating the customer record for it. I suppose we could handle it one of two ways, this way, where we just ignore that its missing the userid and link based on email (which I think is fine), or we could also listen for `customer.updated` and if `userid` is being added in metadata, then add the local record. This approach here seems fine to me.